python/segmentation/models/model_pspnet.py

```diff
@@ -0,0 +1,119 @@
+# DL library imports
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+from torchvision.models import resnet50
+
+
+class auxiliary_branch(nn.Module):
+    """
+    Auxiliary branch helps in setting initial values for the residual blocks.
+    Auxiliary branch is only used during training and not during inference!
+    Auxiliary branch uses similar classifier and loss function of the main branch.
+    TotalLoss=α∗(auxloss)+(1−α)∗(mainloss)
+    Alpha is a hyperparameter. Authors used `alpha - 0.4*
+    """
+    def __init__(self, in_channels, num_classes):
+        super(auxiliary_branch, self).__init__()
+        self.aux = nn.Sequential(
+                    nn.Conv2d(in_channels , 256, kernel_size=3, padding=1, bias=False),
+                    nn.BatchNorm2d(256),
+                    nn.ReLU(inplace=True),
```

Comment:
I think at some point it could be dynamic as well because how do we actually know that this activation function would be the best? Usually it is :) but there is not deterministic since we did not performed any tests :) 

Actually I love to have one day a topic, what determines datasciencists to use actually this type of model archiecture and parameters since there are many ways to propose the shape of the neural network :)